src/components/AuthForm.js

```diff
@@ -363,18 +366,49 @@ export default class AuthForm extends PureComponent<
 
   constructor(props: Object) {
     super(props);
-    const { settings: { authMethods } } = this.props;
-    const defaultAuth = authMethods[0];
     const { schema, uiSchema } = authSchemas[defaultAuth];
     this.state = {
       schema,
       uiSchema,
       formData: { authType: defaultAuth },
     };
+
+    const { history, getServerInfo } = this.props;
+    const server = (history && history[0]) || defaultKintoServer;
+    const anonymousAuth = { authType: "anonymous", server: server };
+
+    getServerInfo(anonymousAuth);
   }
 
+  getSupportedAuthMethods = () => {
+    const { session: { serverInfo: { capabilities } } } = this.props;
+    // Check which of our known auth implementations are supported by the server.
+    const knownAuthMethods = [
+      "basicauth",
+      "account",
+      "fxa",
+      "ldap",
+      "portier",
+      "anonymous",
+    ];
+    let supportedAuthMethods = [defaultAuth];
+    knownAuthMethods.forEach(authMethod => {
+      if (authMethod in capabilities) {
+        supportedAuthMethods.push(authMethod);
+      }
+    });
+    return supportedAuthMethods;
+  };
+
   onChange = ({ formData }: { formData: Object }) => {
-    const { authType } = formData;
+    const { authType, server } = formData;
+    if (this.state.formData.server !== server) {
+      // Server changed, request its capabilities to check what auth methods it
+      // supports.
+      const anonymousAuth = { authType: "anonymous", server: server };
```

Comment:
could use const anonymous type maybe

Reply:
What do you mean?

Reply:
Something like `const ANONYMOUS_AUTH = "anonymous"` ... but that's kind of redundant with the other comments regarding this anonymous string

Reply:
Ok, I added a ANONYMOUS_AUTH_DATA utility.